### PR TITLE
Check if connection name was already set when storing it

### PR DESCRIPTION
**Pull request by wallyqs**
Signed-off-by: Waldemar Quevedo <wally@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/client.go [hunk 1]
```diff
@@ -1883,7 +1883,7 @@ func (c *client) processConnect(arg []byte) error {
 			}
 		}
 		if ncs != _EMPTY_ {
-			c.ncs.Store(fmt.Sprintf("%s - %q", c, ncs))
+			c.ncs.CompareAndSwap(nil, fmt.Sprintf("%s - %q", c, ncs))
```
Comment (derekcollison) [inline server/client.go:1886]: This Go 1.19 only?

Comment (wallyqs) [reply server/client.go:1886]: Looks like built-in since Go 1.17: https://pkg.go.dev/sync/atomic#Value.CompareAndSwap

Comment (derekcollison) [reply server/client.go:1886]: Does this change pass the fuzz test?

Comment (wallyqs) [reply server/client.go:1886]: yes this would fix it

**Review by derekcollison — APPROVED** [review]
LGTM!

**Pull request merged**